src/ngraph/frontend/onnx_import/op/softmax.cpp

```diff
@@ -17,27 +17,92 @@
 #include <memory>
 
 #include "default_opset.hpp"
+#include "ngraph/builder/reshape.hpp"
 #include "ngraph/validation_util.hpp"
 #include "softmax.hpp"
 
 namespace ngraph
 {
     namespace onnx_import
     {
+        namespace
+        {
+            std::shared_ptr<ngraph::Node> onnx_softmax(const std::shared_ptr<ngraph::Node> data,
+                                                       const int64_t axis)
+            {
+                const auto coerced_data = ngraph::builder::opset1::flatten(data, axis);
+
+                const auto axis_1 = default_opset::Constant::create(element::i64, Shape{1}, {1});
+                const auto max = std::make_shared<default_opset::ReduceMax>(coerced_data, axis_1);
+
+                // equivalent to numpy's max.reshape((-1,1))
+                const auto reshape_pattern =
+                    default_opset::Constant::create(element::i64, Shape{2}, {0, 1});
+                const auto reshaped_max =
+                    std::make_shared<default_opset::Reshape>(max, reshape_pattern, true);
+
+                const auto data_minus_max =
+                    std::make_shared<default_opset::Subtract>(coerced_data, reshaped_max);
+
+                const auto exp = std::make_shared<default_opset::Exp>(data_minus_max);
+                const auto sum_exp = std::make_shared<default_opset::ReduceSum>(exp, axis_1);
+                const auto reshaped_sum_exp =
+                    std::make_shared<default_opset::Reshape>(sum_exp, reshape_pattern, true);
+
+                const auto result = std::make_shared<default_opset::Divide>(exp, reshaped_sum_exp);
```

Comment:
You don't need to create subgraph for Softmax here on your own. You may leverage exisiting operator:
```suggestion
const auto result = std::make_shared<default_opset::Softmax>(data_minus_max, 1);
```
I wonder whether this would enable to release the requirement of static input data rank and simplify op logic.

Reply:
Known rank is required to correctly handle 0D and 1D (single op instead of a subgraph). Additionally the static rank requirement also exists in `ngraph::normalize_axis` internally.